### DISCUSSION TITLE
neue Device App Shelly neue App Shelly

### DISCUSSION THREAD

**Discussion by dennisbausch** [camueller/SmartApplianceEnabler#274] [Installation und Betrieb]
Hallo zusammen , ich habe bei mir einen Shelly gehen eine neue Version ausgetauscht  und bekomme jetzt keine werte mehr. 

Alte: Shelly 1pm
neu: Shelly Plus 1pm

Die neunen Plus Shellys setzen auf eine neue API und brauche mal eure Hilfe .  

Alte Zähler Einstellung:
URL: http://192.168.1.XXX/status

Regex: .*[^over]power.:(\d+).*

Wer hat on euch eventuell einen neuen bereits am laufen oder kann mir helfen. 

Hier ist die API Dokumentiert:
https://shelly-api-docs.shelly.cloud/gen2/
Vielleicht  könnten wir dadurch auch die Dokumentation der Shelly  um die neuen Geräte ergänzen. 

Danke

**Comment by dennisbausch** [comment]
https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Shelly_DE.md

**Comment by camueller** [comment]
Gib doch mal in den Web-Browser ein `http://192.168.33.1/rpc/Cover.GetStatus?id=0` und poste, was zurückommt (IP-Adresse natürlich anpassen vorher)
Laut [Doku](https://shelly-api-docs.shelly.cloud/gen2/Components/FunctionalComponents/Cover) müsste da folgendes zurückkommen:
```
{
  "id": 0,
  "source": "WS_in",
  "state": "stopped",
  "apower": 0.0,
  "voltage": 220.4,
  "current": 0.000,
  "pf": 0.00,
  "aenergy": {
    "total": 1.195,
    "by_minute": [
      0.000,
      0.000,
      0.000
    ],
    "minute_ts": 1642086094
  },
  "temperature": {
    "tC": 55.8,
    "tF": 132.4
  },
  "pos_control": true,
  "current_pos": 96
}
```

**Reply by dennisbausch** [reply]
http://192.168.1.207/rpc/Cover.GetStatus?id=0

`
{"code":404,"message":"No handler for Cover.GetStatus"}`


http://192.168.1.207/rpc/Sys.GetStatus

`{"mac":"44179394C118","restart_required":false,"time":"19:06","unixtime":1653757614,"uptime":15263,"ram_size":239600,"ram_free":167116,"fs_size":458752,"fs_free":221184,"cfg_rev":11,"available_updates":{"stable":{"version":"0.10.1"}}}`

**Comment by camueller** [comment]
Probier mal http://192.168.33.1/rpc/Switch.GetStatus?id=0
da sollte das hier zurückkommen:
```
{
   "id": 0,
   "source": "timer",
   "output": false,
   "timer_started_at": 1626942399.36,
   "timer_duration": 60,
   "apower": 0.0,
   "voltage": 237.5,
   "current": 0.0,
   "pf": -0.41,
   "aenergy": {
      "total": 14.093,
      "by_minute": [
         0,
         88.547,
         47.276
      ],
      "minute_ts": 1626942456
   },
   "temperature": {
      "tC": 57,
      "tF": 134.6
   },
   "errors": ["overpower", "overvoltage"]
}
```

**Reply by dennisbausch** [reply]
`{"id":0, "source":"init", "output":true, "apower":6.1, "voltage":224.6, "current":0.057, "aenergy":{"total":7245.013,"by_minute":[61.245,102.270,102.109],"minute_ts":1653767195},"temperature":{"tC":54.9, "tF":130.8}}`


apower":6.1

sieht doch schon sehr gut aus !

**Reply by dennisbausch** [reply]
Damit sollte es gehen oder einen besseren Vorschlag :


`.*[^over]apower.:(\d+).*`

**Reply by camueller** [reply]
Feld | Wert
-- | --
Format | JSON
URL | http://192.168.33.1/rpc/Switch.GetStatus?id=0
Pfad für Extraktion (bei Parameter Leistung) | $.apower

Keine `Regex für Extraktion` angeben!

**Reply by dennisbausch** [reply]
Danke das klappt. 
würdest du die Shelly Doku entsprechend anpassen ?

**Comment by dennisbausch** [comment]
Alte und neue API 

Wäre es möglich das du auch die Statusabfrage für die alte und neue API mit in die Doku aufnehmen:

Alte API
URL: http://192.168.1.206/relay/0/status

ON
`{"ison":true,"has_timer":false,"timer_started":0,"timer_duration":0,"timer_remaining":0,"overpower":false,"source":"http"}`
OF
`{"ison":false,"has_timer":false,"timer_started":0,"timer_duration":0,"timer_remaining":0,"overpower":false,"source":"http"}
`

**Welcher Regex  ?** 

Neu API
URL:
http://192.168.1.207/rpc/Switch.GetStatus?id=0

ON
`{"id":0, "source":"init", "output":true, "apower":6.2, "voltage":222.6, "current":0.057, "aenergy":{"total":48.271,"by_minute":[27.122,101.873,101.810],"minute_ts":1653816494},"temperature":{"tC":47.1, "tF":116.9}}`

OFF
`{"ison":true,"has_timer":false,"timer_started":0,"timer_duration":0,"timer_remaining":0,"overpower":false,"source":"http"}`

**Welcher Regex  ?**
